### ISSUE TITLE
Merge command shortcut

### ISSUE THREAD

**Issue by lazywei** [github/hub#786]
Hi,

I'm just wondering if I must type the full PR url (i.e., https://github.com/me/my-repo/pull/1) when using git merge? Is it possible for me to only type, e.g., `git merge pull/1`?
Thanks.

**Comment by jingweno** [comment]
The branch name is ambiguous. It's not uncommon a branch is named as `xxx/xxx`. We wouldn't know if the user wants to merge a PR or a branch.

**Comment by mislav** [comment]
@lazywei That's a good idea and we'd like to support it but, as @jingweno said, we wouldn't be able to distinguish between a PR reference and a branch that could be named the same way in your local repository.

We would have to come up with a better syntax. We tried `#123` before but it wasn't such a good idea since `#` is a comment character in bash.

Closing because this is [on our roadmap](https://trello.com/b/5MKBrz9K/github-cli) already. Thanks for the suggestion!
